posts/templatetags/parser.py

```diff
@@ -0,0 +1,24 @@
+"""@package docstring
+Extra template functions
+
+@author 7Pros
+@copyright
+"""
+from django import template
+from django.core.urlresolvers import reverse
+from posts.views import PostsListView
+from django.template.defaultfilters import stringfilter
+import re
+
+register = template.Library()
+
+@register.filter
+@stringfilter
+def parse(content):
+    pattern = re.compile(r"#(\w+).*?", flags=re.IGNORECASE | re.UNICODE)
+    parsed_content = pattern.sub(createHashtagURL, content)
+
+    return parsed_content
```

Comment:
This filter does not work for special chars that get HTML-escaped, at least `&'"<>` and probably [even more](https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references#Character_entity_references_in_HTML).

When posting `I'm here` for example, the result gets rendered as `I&#39;m here`, with the `#39` highlighted as a hashtag.

The corresponding HTML looks like this:
```html
I&amp;
<a class="hashtag" href="/posts/show/39/">#39</a>
;m here
```

This is seems to be the result of searching the escaped text (`I&#39;m here`, seems to have the hashtag `#39`) for hashtags instead of the actually typed text (`I'm here`, has no hashtags).